src/registry.js

```diff
@@ -0,0 +1,296 @@
+"use strict";
```

Comment:
How much of this is the original real polyfill, and how much is extensions on top of it?

If at all possible (and I do get that it might not be) I'd like to keep them separate. It's going to be way easier to maintain this if there's a polyfill file that we can trust (and upgrade) independently, and then a separate bunch of code making the any tweaks required to hook it into Domino, and/or wrapping the polyfill to change the interface.